src/pipeline-stages/src/test/scala/TextPreprocessorSuite.scala

```diff
@@ -0,0 +1,133 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.ml.spark
+
+import org.apache.spark.ml.util.MLReadable
+
+class TextPreprocessorSuite extends TestBase with TransformerFuzzing[TextPreprocessor] {
+  val toMap1 = "The happy sad boy drank sap"
+  val toMap2 = "The hater sad doy drank sap"
+  val toMap3 = "The hater sad doy"
+  val wordDF = session.createDataFrame(Seq(
+    (0, toMap1, "The sad sap boy drank sap"),
+    (1, toMap2, "The sap sap drank sap"),
+    (2, "foo", "foo"),
+    (3, toMap3 + "aABc0123456789Zz_", "The sap sap")))
+    .toDF("label", "words1", "words2")
+
+  val testMap = Map[String, String] (
+    "happy"   -> "sad",
+    "hater"   -> "sap",
+    "sad"     -> "sap",
+    "sad doy" -> "sap"
+  )
+
+  val testTrie1 = new Trie().put("happy", "sad").put("hater", "sap")
+  var testTrie1Pivot: Trie = testTrie1
+  for (letter <- "ha") testTrie1Pivot = testTrie1Pivot.get(letter).get
+
+  test("Check basic string mapper") {
+    {
+      var copyHappy: Trie = testTrie1
+      for (letter <- "happy") copyHappy = copyHappy.get(letter).get
+      assert(copyHappy != null)
+      assert(copyHappy.value.mkString("") == "sad")
+    }
+
+    {
```

Comment:
Add a new test case for each instead of adding one big test. This will allow you to track failures in a more granular manner.